android/src/com/frostwire/android/gui/activities/MainActivity.java

```diff
@@ -1001,4 +1016,42 @@ private static String saveViewContent(Context context, Uri uri, String name) {
 
         return "file://" + target.getAbsolutePath();
     }
+
+
+    public void swapDrawerForBack() {
+        drawerToggle.setDrawerIndicatorEnabled(false);
+        ActionBar actionBar = getActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayShowHomeEnabled(false);
+            actionBar.setDisplayShowTitleEnabled(false);
+        }
+
+        drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+        updateHeader(getCurrentFragment());
+    }
+
+    private void swapBackForDrawer() {
+        drawerToggle.setDrawerIndicatorEnabled(true);
+        ActionBar actionBar = getActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayShowHomeEnabled(true);
+            actionBar.setDisplayShowTitleEnabled(true);
+        }
+
+        drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
+        updateHeader(getCurrentFragment());
+    }
```

Comment:
these 2 methods should probably be a single one with a boolean parameter:
```java
/**
 * @param drawerForBack - if true, swaps drawer for back, otherwise back for drawer
 */
public void swapDrawerForBack(boolean drawerForBack) { 
    drawerToggle.setDrawerIndicatorEnabled(!drawerForBack);
    ActionBar actionBar = getActionBar();
    if (actionBar != null) {
        actionBar.setDisplayShowHomeEnabled(!drawerForBack);
        actionBar.setDisplayShowTitleEnabled(!drawerForBack);
    }
   drawerLayout.setDrawerLockMode(drawerForBack ? DrawerLayout.LOCK_MODE_LOCKED_CLOSED : DrawerLayout.LOCK_MODE_UNLOCKED);
   updateHeader(getCurrentFragment());
}
```